src/client.rs

```diff
@@ -149,13 +151,15 @@ impl ChatGPT {
         &self,
         history: &Vec<ChatMessage>,
     ) -> crate::Result<impl Stream<Item = ResponseChunk>> {
+        use std::str::FromStr;
+
         use eventsource_stream::Eventsource;
         use futures_util::StreamExt;
 
         let response_stream = self
             .client
             .post(
-                Url::from_str(self.config.api_url)
+                Url::parse(self.config.api_url.as_str())
                     .map_err(|err| crate::err::Error::ParsingError(err.to_string()))?,
             )
```

Comment:
This can be changed to:
```suggestion
            .post(self.config.api_url.clone())
```